File: solutions/java/matrix/1/src/main/java/Matrix.java
import java.util.stream.Stream;

class Matrix {

    int[][] rows;
    int[][] cols;

    Matrix(String matrixAsString) {

        rows = Stream.of(matrixAsString.split("\\n"))
                .map(row -> buildRow(row))
                .toArray(int[][]::new);

        buildCols();
    }

    private int[] buildRow(String row) {
        return Stream.of(row.split(" "))
                .mapToInt(Integer::parseInt)
                .toArray();
    }

    private void buildCols() {
        cols = new int[rows[0].length][rows.length];

        for (int j = 0; j < rows[0].length; j++) {
            for (int i = 0; i < rows.length; i++) {
                cols[j][i] = rows[i][j];
            }
        }
    }

    int[] getRow(int rowNumber) {
        return rows[rowNumber - 1];
    }

    int[] getColumn(int columnNumber) {
        return cols[columnNumber - 1];
    }
}

File: solutions/typescript/matrix/1/matrix.ts
class Matrix {
  private matrix: number[][];

  constructor(matrix: string) {
    // Your code here
    this.matrix = matrix
      .split("\n")
      .map((n) => n.split(" ").map((n) => parseInt(n)));
  }

  get rows() {
    return this.matrix;
  }

  get columns() {
    return this.matrix.reduce((acc: number[][], val, i) => {
      if (i === 0 && acc.length === 0) {
        acc = Array.from(Array(val.length), () => []);
      }

      for (const idx in val) {
        acc[idx][i] = val[idx];
      }

      return acc;
    }, []);
  }
}

export default Matrix;
